extensions/agent/src/mainController.ts

```diff
@@ -91,6 +124,134 @@ export class MainController {
 			}
 			this.proxyDialog.dialogName ? await this.proxyDialog.openDialog(this.proxyDialog.dialogName) : await this.proxyDialog.openDialog();
 		});
+
+		vscode.commands.registerCommand('agent.openNotebookEditorFromJsonString', async (filename: string, jsonNotebook: string, notebookInfo?: azdata.AgentNotebookInfo, ownerUri?: string) => {
+			const tempfilePath = path.join(os.tmpdir(), 'mssql_notebooks', filename + '.ipynb');
+			if (!await promisify(fs.exists)(path.join(os.tmpdir(), 'mssql_notebooks'))) {
+				await promisify(fs.mkdir)(path.join(os.tmpdir(), 'mssql_notebooks'));
+			}
+			let editors = azdata.nb.visibleNotebookEditors;
+			if (await promisify(fs.exists)(tempfilePath)) {
```

Comment:
Since you use this multiple times I would just do the promisify once as a const outside this class or in a utils file and just refer to it there.